rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMService.java

```diff
@@ -118,6 +132,25 @@ String defineNodeSource(String sessionId, String nodeSourceName, String infrastr
             String[] policyParameters, String[] policyFileParameters, String nodesRecoverable)
             throws RestServerException, ServiceException;
 
+    /**
+     * Edit a NodeSource
+     * @param sessionId current session
+     * @param nodeSourceName name of the NS to edit
+     * @param infrastructureType infrastructure manager full class name
+     * @param infrastructureParameters IM String parameters, null value for files
+     * @param infrastructureFileParameters file parameters
+     * @param policyType policy full class name
+     * @param policyParameters String parameters, null value for files
+     * @param policyFileParameters file parameters
+     * @param nodesRecoverable whether nodes can be recovered after a crash
+     * @throws RestServerException
+     * @throws ServiceException
+     */
+    String editNodeSource(String sessionId, String nodeSourceName, String infrastructureType,
+            String[] infrastructureParameters, String[] infrastructureFileParameters, String policyType,
```

Comment:
Is it possible to use an object/create a new classe to have less parameters?

Reply:
Unfortunately no, this matches the REST API of the Resource Manager backend.